File: 205_isomorphic_strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Object, Integer> m = new HashMap<>();
        for (Integer i = 0; i < s.length(); i++) {
            // t.chatAt(i) + "" is for different object type
            // therefore, if we have the same character, we still get null
            Integer sValue = m.put(s.charAt(i), i);
            Integer tValue = m.put(t.charAt(i) + "", i);
            if (sValue != tValue) {
                return false;
            }
        }
        return true;
    }
}

File: python/205_isomorphic_strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        s_map, t_map = dict(), dict()
        if len(s) != len(t):
            return False
        for i in range(0, len(s)):
            if (s[i] in s_map and s_map[s[i]] != t[i]) or (t[i] in t_map and t_map[t[i]] != s[i]):
                return False
            else:
                s_map[s[i]] = t[i]
                t_map[t[i]] = s[i]
        return True

